### ISSUE TITLE
Part 5. Classification

### ISSUE THREAD

**Issue by XuSShuai** [XuSShuai/Machine-Learning-2017-Fall#5]
## Probabilistic Generative Model

 - credit scoring
 - medical diagnosis
 - handwritten character recognition
 - face recognition

## Example Application

任务：根据宝可梦的属性，预测宝可梦的类别。

## How to do Classification?

 - Training data for Classification
将编号400以下的作为训练集，400以后的作为测试集，训练模型来预测以后产生的宝可梦的类别。
    
 - Classification as Regression for binary task?
可以利用回归模型来解决二类分类问题吗？将类别1的target value设定为1；将类别2的target value设定为-1，在这样的数据上训练一个regression模型。在测试阶段，如果output大于0，那么为类别1；如果output小于0，那么为类别2。
这并不是一个好的做法：对于下图左边的数据似乎是没有什么问题的，**但是**如果数据的分布如下图右，那么一个好的classifier应该如绿线所示，但是利用regression model得到的结果会如蓝线所示，因为在regression model看来，右下角的那些蓝色数据点不是confidence很高的、被正确划分的样本点（在classification model看来是这样），反而是error，因为对于这些点，回归模型计算得target value远远大于1，这会使得mean square error变大。从而regression model会将分类边界从绿线偏向蓝线以获得更小的error，而这显然不是一个好的结果。
**penalize to the examples that are too correct.**
另外对于多分类问题，如果也当做regression task来做的话，其类标签之间会被隐式的添加不存在距离关系。
![image](https://user-images.githubusercontent.com/29543183/48965758-3e415200-effe-11e8-9637-931799bd6b3b.png)

## Ideal Alternatives

![image](https://user-images.githubusercontent.com/29543183/48965968-0d175080-f003-11e8-931e-7a5907c45f48.png)
loss function不是可微分的，所以不可以用gradient descent来进行求解，但是SVM，感知器可以用于解决这类问题。

## Two Boxs

![image](https://user-images.githubusercontent.com/29543183/48966011-a5153a00-f003-11e8-8989-884b48624528.png)
随机抽取一个篮球，来自盒子1的概率$P(B_1|Blue)$的计算公式如下：
$$P(B_1|\text{Blue}) = \frac{P(B_1)P(\text{Blue}|B_1)}{P(B_1)P(\text{Blue}|B_1) + P(B_2)P(\text{Blue}|B_2)}$$

## Two Class

![image](https://user-images.githubusercontent.com/29543183/48966071-7055b280-f004-11e8-960f-e30b755e1403.png)

对于给定的x，其属于类别1的概率：
$$P(C_1|x) = \frac{P(C_1)P(x|C_1)}{P(C_1)P(x|C_1) + P(C_2)P(x|C_2)}$$
这样的模型称为$\text{generative model}$，因为我们可以知道某一个x产生的概率：
$$P(x) = P(C_1)P(x|C_1) + P(C_2)P(x|C_2)$$
